client/catapult/CMakeGlobalSettings.cmake

```diff
@@ -244,9 +244,16 @@ function(catapult_set_test_compiler_options)
 	# some gtest workarounds for gcc + clang
 	if("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
 		# - Wno-dangling-else: workaround for GTEST ambiguous else blocker not working https://github.com/google/googletest/issues/1119
-		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} \
-			-Wno-dangling-else"
-			PARENT_SCOPE)
+		# disable dangling reference for tests - https://gcc.gnu.org/bugzilla/show_bug.cgi?id=108165#c9
+        if("${CMAKE_CXX_COMPILER_VERSION}" MATCHES "^13.")
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} \
+                -Wno-dangling-else -Wno-dangling-reference"
+                PARENT_SCOPE)
+        else()
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} \
+                -Wno-dangling-else"
```

Comment:
i guess you don't need to disable this warning for 13 ? 🤔

Reply:
It is still needed.  this is the option for gcc13
``-Wno-dangling-else -Wno-dangling-reference``